components/binding/internal/storage/error.go

```diff
@@ -0,0 +1,35 @@
+package storage
```

Comment:
maybe should be in `internal/error/error.go`, then we can use it in other places without import package "storage", besides maybe all errors should be in one place, what do you think?

Reply:
I think i can move that implementation to the error package why not 👍

Reply:
I'll also change the name of package to `errors` because error conflicts with the go type name